Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -189,6 +189,76 @@ extension NIOSSLCertificate {
 
         return NIOSSLPublicKey.fromInternalPointer(takingOwnership: key)
     }
+
+    /// Create an array of `NIOSSLCertificate`s from a buffer of bytes in PEM format.
+    ///
+    /// - Parameter buffer: The PEM buffer to read certificates from.
+    /// - Throws: If an error is encountered while reading certificates.
+    public class func fromPEMBuffer(_ buffer: [Int8]) throws -> [NIOSSLCertificate] {
+        CNIOBoringSSL_ERR_clear_error()
+        defer {
+            CNIOBoringSSL_ERR_clear_error()
+        }
+
+        return try buffer.withUnsafeBytes { (ptr) -> [NIOSSLCertificate] in
+            let bio = CNIOBoringSSL_BIO_new_mem_buf(UnsafeMutableRawPointer(mutating: ptr.baseAddress!), Int32(ptr.count))!
+            defer {
+                CNIOBoringSSL_BIO_free(bio)
+            }
+
+            return try readCertificatesFromBIO(bio)
+        }
+    }
+
+    /// Create an array of `NIOSSLCertificate`s from a file at a given path in PEM format.
+    ///
+    /// - Parameter file: The PEM file to read certificates from.
+    /// - Throws: If an error is encountered while reading certificates.
+    public class func fromPEMFile(_ path: String) throws -> [NIOSSLCertificate] {
+        CNIOBoringSSL_ERR_clear_error()
+        defer {
+            CNIOBoringSSL_ERR_clear_error()
+        }
+
+        guard let bio = CNIOBoringSSL_BIO_new(CNIOBoringSSL_BIO_s_file()) else {
+            throw NIOSSLError.unableToAllocateBoringSSLObject
+        }
+        defer {
+            CNIOBoringSSL_BIO_free(bio)
+        }
+
+        try path.utf8CString.withUnsafeBufferPointer {
+            guard let baseAddress = $0.baseAddress, CNIOBoringSSL_BIO_read_filename(bio, baseAddress) > 0 else {
+                throw NIOSSLError.failedToLoadCertificate
+            }
+        }
```

Comment:
This can probably be simplified: you should be able to just do:

```swift
guard CNIOBoringSSL_BIO_read_filename(bio, &path) > 0 else {
    throw NIOSSLError.failedToLoadCertificate
}
```

Reply:
Neat, I didn't realise this was possible! (No `&` required though)